src/shared/dissect-image.c

```diff
@@ -450,16 +469,21 @@ static int wait_for_partition_device(
                         return r;
         }
 
-        r = sd_event_add_time_relative(
-                        event, &retry_source,
-                        CLOCK_MONOTONIC, 500 * USEC_PER_MSEC, 0,
-                        retry_handler, &w);
-        if (r < 0)
-                return r;
+        /* If we have a disk sequence number we can do exact matching, and
+         * given they are monotonic we know if we missed it or if we can still
+         * find it. */
```

Comment:
i'd probably reverse the comment, explaining why we do the retry thing if we have no diskseq.